qiskit_braket_provider/providers/adapter.py

```diff
@@ -531,14 +529,17 @@ def to_qiskit(circuit: Circuit) -> QuantumCircuit:
     if not isinstance(circuit, Circuit):
         raise TypeError(f"Expected a Circuit, got {type(circuit)} instead.")
 
-    qiskit_circuit = QuantumCircuit(circuit.qubit_count)
+    num_measurements = sum(
+        instr.operator.name.lower() == "measure" for instr in circuit.instructions
```

Comment:
Why a name check rather than `isinstance(instr.operator, Measure)`?

Reply:
Good point! An isinstance check is probably a safer way to go. Updated to use `isinstance(instr.operator, Measure)` instead.